### PR TITLE
feat: add package @alg-wiki/runner and python3 language

### PR DESCRIPTION
**Pull request by acheronfail**
Something I've noticed while working with the monorepo is that I can't use other packages when running scripts, without first building it.

A little frustrating to the dev-loop, since each time I try to run something, I need to remember when the last time that I built the monorepo was, and if I've changed a package other than the one I'm running, etc.

It would be nice if we could set it up so when running `ts-node`, etc, it just reads from all the source files...

### CODE REVIEW

File: packages/runner/src/result-emitter.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import { Boundary } from "./boundary";
+
+export enum ResultEmitterEvent {
+  Result = "result",
+  Stdout = "stdout",
+  Stderr = "stderr",
+  Close = "close",
+}
+
+// TODO: type-safety when adding listeners (get callbacks to have the right types/args/etc)
```
Comment (acheronfail) [inline packages/runner/src/result-emitter.ts:20]: I couldn't figure out a way to get nice typed callback signatures. I didn't try that hard either 😆 but if you have any ideas shout out.

File: packages/runner/src/scripts/build.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { execSync } from "child_process";
```
Comment (acheronfail) [inline packages/runner/src/scripts/build.ts:1]: Had to add this file and the one below it into `src`, even though they're just scripts.
Could change the `tsconfig.json`, but perhaps we want to think about some sort of convention?

These aren't meant to be source files, but scripts to be run during development, etc.

File: packages/website/package.json [hunk 3]
```diff
@@ -28,13 +28,15 @@
...
   "dependencies": {
+    "@alg-wiki/types": "workspace:packages/types",
     "@codemirror/basic-setup": "^0.19.0",
     "@codemirror/commands": "^0.19.3",
     "@codemirror/lang-javascript": "^0.19.1",
     "@codemirror/state": "^0.19.1",
     "@codemirror/view": "^0.19.6",
     "@material-ui/core": "^4.12.3",
     "@material-ui/icons": "^4.11.2",
+    "@tiptap/core": "^2.0.0-beta.106",
```
Comment (acheronfail) [inline packages/website/package.json:39]: This was a missing peer dependency

File: packages/runner/src/boundary.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { randomBytes } from "crypto";
+import { promisify } from "util";
+
+export class Boundary {
+  public readonly error: string;
+  public readonly start: string;
+  public readonly end: string;
+
+  public static async create(): Promise<Boundary> {
+    const buf = await promisify(randomBytes)(16);
```
Comment (jakzo) [inline packages/runner/src/boundary.ts:10]: Not sure how long randomBytes takes but I think I already made a synchronous random string generator somewhere in this repo and we don't need it to be cryptographically secure

Comment (acheronfail) [reply packages/runner/src/boundary.ts:10]: > we don't need it to be cryptographically secure

Are you sure? Thinking about alf.nu/returnTrue's `random4` example 😆

File: packages/runner/src/result-emitter.ts (hunk 1 above)
Comment (jakzo) [reply packages/runner/src/result-emitter.ts:20]: Another TODO is to limit the amount of output to a max size

**Comment by acheronfail** [comment]
Closing this in favour of a new branch/implementation. :)

**Pull request closed without merging**